### PR TITLE
vr: cleanup files in /var/cache/cloud/processed every day

### LINKED ISSUE #3757: Files in /var/cache/cloud/processed on virtual router are not cleaned up
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
System VMs
~~~

##### CLOUDSTACK VERSION

~~~
4.13.0
~~~

##### CONFIGURATION

Advanced networking

##### OS / ENVIRONMENT

RHEL 7.x

##### SUMMARY

We are running some tests that involve creating and deleting lots of instances using the API. Our virtual router ran out of inodes and had over 100,000 files in the /var/cache/cloud/processed directory. I can't find any code in CloudStack that cleans up that directory, but if there is then there should also be an option to cleanup more frequently.

##### STEPS TO REPRODUCE

Create/delete thousands of VMs.

##### EXPECTED RESULTS

I'd expect the virtual router to clean up after itself.

##### ACTUAL RESULTS

The virtual router eventually runs out of inodes.

~~~
$ df -i /var
Filesystem     Inodes  IUsed IFree IUse% Mounted on
/dev/vda6      117392 117392     0  100% /var

$ ls -f /var/cache/cloud/processed | wc -l
110343
~~~

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes #3757 

This adds a daily cron job to clean up files in /var/cache/cloud/processed older than 180 days, 179 days, ... until the disk space usage does not exceed threshold (80%). the logs in last 5 days will not be cleaned.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Review by nvazquez — APPROVED** [review]
Code LGTM

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by ravening** [comment]
@weizhouapache the logrotate service should be able to take care of this right rather than adding a new shell script? we can specify the duration to keep the logs also

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
> @weizhouapache the logrotate service should be able to take care of this right rather than adding a new shell script? we can specify the duration to keep the logs also

@ravening no. logrorate will rotate the log files in /var/log

**Comment by ravening** [comment]
> > @weizhouapache the logrotate service should be able to take care of this right rather than adding a new shell script? we can specify the duration to keep the logs also
> 
> @ravening no. logrorate will rotate the log files in /var/log

@weizhouapache but we can add an extra rule to monitor /var/cache also right?

**Comment by weizhouapache** [comment]
> > > @weizhouapache the logrotate service should be able to take care of this right rather than adding a new shell script? we can specify the duration to keep the logs also
> > 
> > 
> > @ravening no. logrorate will rotate the log files in /var/log
> 
> @weizhouapache but we can add an extra rule to monitor /var/cache also right?

@ravening the problem is, we do not know what the filenames are. 
here is an example
```
root@r-88-VM:~# ls -l /var/cache/cloud/processed/
total 24
-rw-r--r-- 1 root root 239 Aug 19 07:44 vm_dhcp_entry.json.82cf8311-f5d4-4a10-9b50-001e0558082a.gz
-rw-r--r-- 1 root root 239 Aug 19 07:42 vm_dhcp_entry.json.ef071140-60d8-4a20-970d-ca4979aee9f5.gz
-rw-r--r-- 1 root root 406 Aug 19 07:44 vm_metadata.json.26853b78-280b-4ea5-94b2-55e83bb332e4.gz
-rw-r--r-- 1 root root 406 Aug 19 07:42 vm_metadata.json.27cbd5c6-43b8-417f-8c1d-e51fc697ce10.gz
-rw-r--r-- 1 root root 160 Aug 19 07:42 vm_password.json.08296973-3f9a-4fb5-ac75-d9c9c9c9c8fb.gz
-rw-r--r-- 1 root root 160 Aug 19 07:44 vm_password.json.dea5c53d-043a-43f1-980c-f904a1e7b5b7.gz
```

**Review by davidjumani — APPROVED** [review]
CLGTM

File: systemvm/debian/etc/cron.daily/clean_cloud_processed_files [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (DaanHoogland) [inline systemvm/debian/etc/cron.daily/clean_cloud_processed_files:1]: license header needed?

Comment (weizhouapache) [reply systemvm/debian/etc/cron.daily/clean_cloud_processed_files:1]: @DaanHoogland good question...

Comment (weizhouapache) [reply systemvm/debian/etc/cron.daily/clean_cloud_processed_files:1]: @DaanHoogland done.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Pull request merged**